src/rpc/tts.rs

```diff
@@ -0,0 +1,159 @@
+    }
+}
+
+#[tonic::async_trait]
+impl TtsService for Tts {
+    async fn transmit(
+        &self,
+        request: Request<tts::v0::TransmitRequest>,
+    ) -> Result<Response<tts::v0::TransmitResponse>, Status> {
+        let request = request.into_inner();
+        let name = request.name.as_deref().unwrap_or("DCS-gRPC");
+        let mut client = srs::Client::new(
+            name,
+            request.frequency,
+            match Coalition::from_i32(request.coalition) {
+                Some(Coalition::Red) => srs::Coalition::Red,
+                _ => srs::Coalition::Blue,
+            },
+        );
+        let position = request.position.unwrap_or_default();
+        client
+            .set_position(srs::Position {
+                lat: position.lat,
+                lon: position.lon,
+                alt: position.alt,
+            })
+            .await;
+
+        let addr = SocketAddr::from_str(request.srs_addr.as_deref().unwrap_or("127.0.0.1:5002"))
+            .map_err(|_| Status::invalid_argument("`srs_addr` is not a valid socket address"))?;
+        let stream = client
+            .start(addr, self.shutdown_signal.signal())
+            .await
+            .map_err(|err| Status::internal(err.to_string()))?;
+
+        let config = match request.provider {
+            Some(transmit_request::Provider::Aws(transmit_request::Aws {
+                voice,
+                key,
+                secret,
+                region,
+            })) => TtsConfig::Aws(AwsConfig {
+                voice,
+                key,
+                secret,
+                region: AwsRegion::from_str(&region)
+                    .map_err(|err| Status::internal(err.to_string()))?,
+            }),
+            Some(transmit_request::Provider::Azure(transmit_request::Azure {
+                voice,
+                key,
+                region,
+            })) => TtsConfig::Azure(AzureConfig { voice, key, region }),
+            Some(transmit_request::Provider::Gcloud(transmit_request::GCloud { voice, key })) => {
+                TtsConfig::GCloud(GCloudConfig { voice, key })
+            }
+            Some(transmit_request::Provider::Win(transmit_request::Windows { voice })) => {
+                TtsConfig::Win(WinConfig { voice })
+            }
+            None => TtsConfig::Win(WinConfig { voice: None }),
+        };
+
+        if request.wait {
+            transmit(
+                &request.text,
+                &config,
+                stream,
+                self.shutdown_signal.signal(),
+            )
+            .await
+            .map_err(|err| Status::internal(err.to_string()))?;
+        } else {
+            let signal = self.shutdown_signal.signal();
+            tokio::task::spawn(async move {
+                if let Err(err) = transmit(&request.text, &config, stream, signal).await {
+                    log::error!("TTS transmission failed: {}", err);
+                }
+            });
+        }
```

Comment:
Future improvement: keep SRS client connections open for a bit to re-use them for subsequent requests

Reply:
Created an issue for it #199